packages/next/build/webpack/plugins/build-manifest-plugin.ts

```diff
@@ -63,10 +87,25 @@ function getFilesArray(files: any) {
 export default class BuildManifestPlugin {
   private buildId: string
   private modern: boolean
+  private rewrites: Rewrite[]
 
-  constructor(options: { buildId: string; modern: boolean }) {
+  constructor(options: {
+    buildId: string
+    modern: boolean
+    rewrites: Rewrite[]
+  }) {
     this.buildId = options.buildId
     this.modern = options.modern
+    // TODO: should public files be sent in case they match and the
```

Comment:
This is a tricky one. I probably vote no for now but we should open an issue to track this.

Reply:
Sounds good, opened issue to investigate this further here https://github.com/vercel/next.js/issues/16092